### PR TITLE
fix: Adding code to regenerate validation test files

### LINKED ISSUE #638: Determine if temp validation files need to be regenerated
There is a relatively minor issue where because the temp files generated in order to run `validation/` are cached that can cause local tests to fail. 

This ticket would either
1. Generate new temp files with each test suite run to avoid this mismatch
2. Use a checker of some kind to determine if new temp files are needed

Maybe there's another solution. Not a very high pressure ticket though, given that it is easily resolved by just deleting and rerunning validation.

### Acceptance Criteria

**Scenario 1: Automated detection of stale validation files**
- **Given** the validation test suite is initiated
- **When** the system detects that the existing temporary validation files are out of sync with the current source code or configuration
- **Then** the system should automatically regenerate the necessary temporary files before executing the tests

**Scenario 2: Successful test execution without manual intervention**
- **Given** a local environment with cached temporary validation files that would previously cause a mismatch failure
- **When** the user runs the `validation/` test suite
- **Then** the tests should pass successfully without requiring the user to manually delete the temporary files

**Scenario 3: Optimization of test suite performance (Optional/Conditional)**
- **Given** the temporary validation files are already up-to-date and match the current source
- **When** the validation test suite is executed
- **Then** the system should skip the regeneration step to minimize test execution time (if the "checker" approach is implemented)

### PR DESCRIPTION
**Pull request by robertandremitchell**
## Description

This adds code to `validation/main.py` that checks the last modified date of the two dependent files to see if new temps are needed. This should be testable locally by making a nonsense change (adding a `<>` to the blank line 3 of the `packages/validation/src/validation/eicr-validator/schematron/APHL_TextToCodeSchematron_09252025.sch`) and then confirming that new temp files were generated in `packages/validation/src/validation/eicr-validator/output`.

This also adds a just command to manually remove the files to generate new ones, regardless of whether the files had changed.

These changes are needed to address issues that can arise where locally, the temp files become stale as a result of changes to the schematron or xslt compile file.

## Related Issues

Closes #638, #655 

## Additional Notes

[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: .justscripts/just/test.just [hunk 1]
```diff
@@ -14,4 +14,9 @@ all *ARGS:
 
 [doc("Run tests with coverage, specifying the package to measure coverage for")]
 coverage *ARGS:
-    uv run pytest {{ ARGS }} --cov={{ARGS}} --cov-report=term-missing
\ No newline at end of file
+    uv run pytest {{ ARGS }} --cov={{ARGS}} --cov-report=term-missing
+
+[doc("Regenerate validation temp files and run validation tests")]
+validation:
+    uv run python -c "from pathlib import Path; output = Path('validation/src/validation/eicr-validator/output'); [file.unlink(missing_ok=True) for file in (output / 'stage1.sch.tmp', output / 'stage2.sch.tmp', output / 'validator.xsl.tmp', output / 'voc_ttc.xml')]"
```
Comment (nickclyde) [inline .justscripts/just/test.just]: I think this path is missing the `packages/` prefix. `test.just` sets `working-directory := '../..'` (the repo root), and from there the output dir lives at `packages/validation/src/validation/eicr-validator/output`. The `pytest` line right below already uses the `packages/`-prefixed path. Could we change it to `packages/validation/src/validation/eicr-validator/output`?

File: packages/validation/tests/test_validation.py [hunk 2]
```diff
@@ -258,3 +259,69 @@ def test_build_schematron_report_xml_normalizes_real_location():
...
+
+    old_time = 1_000_000_000
+    new_time = 1_000_000_100
+
+    for output_file in (stage1_output, stage2_output, validator_output, voc_output):
+        output_file.touch()
+        output_file.chmod(0o644)
+        output_file.write_text(output_file.read_text())
+        output_file.stat()
+        output_file.touch()
```
Comment (nickclyde) [inline packages/validation/tests/test_validation.py]: I think this setup loop has some redundant steps. The files already have content written above, and the only operation that affects the test is the final `os.utime(...)` that backdates the mtime. Could we simplify the loop body to just the `os.utime(output_file, (old_time, old_time))` call to make the intent clearer?

**Review by nickclyde — COMMENTED** [review]
Looks good! Just a few notes:

File: packages/validation/tests/test_validation.py (hunk 2 above)
Comment (robertandremitchell) [reply packages/validation/tests/test_validation.py]: agreed. I think this does a better job of simplifying the denseness of this code. lmk!

File: .justscripts/just/test.just (hunk 1 above)
Comment (robertandremitchell) [reply .justscripts/just/test.just]: updated!

**Review by nickclyde — APPROVED** [review]
Thanks Rob!!
